src/core/linepolygonshape.cpp

```diff
@@ -0,0 +1,240 @@
+/***************************************************************************
+              linepolygonshape.cpp
+               ----------------------------------------------------
+              date                 : 21.11.2023
+              copyright            : (C) 2023 by Mathieu Pellerin
+              email                : mathieu at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "linepolygonshape.h"
+
+#include <qgscoordinatereferencesystem.h>
+#include <qgscoordinatetransform.h>
+#include <qgscurve.h>
+#include <qgsgeometry.h>
+#include <qgspolygon.h>
+#include <qgsproject.h>
+
+
+LinePolygonShape::LinePolygonShape( QQuickItem *parent )
+  : QQuickItem( parent )
+{
+  setFlags( QQuickItem::ItemHasContents );
+  setTransformOrigin( QQuickItem::TopLeft );
+  setAntialiasing( true );
+}
+
+void LinePolygonShape::createPolylines()
+{
+  const QgsRectangle visibleExtent = mMapSettings->visibleExtent();
+  const double scaleFactor = 1.0 / mMapSettings->mapUnitsPerPoint();
+
+  mPolylines.clear();
+
+  QgsGeometry geometry( mGeometry ? mGeometry->qgsGeometry() : QgsGeometry() );
+  Qgis::GeometryType geomType = Qgis::GeometryType::Null;
+  if ( mGeometry && !geometry.isEmpty() && geometry.type() != Qgis::GeometryType::Point )
+  {
+    QgsCoordinateTransform ct( mGeometry->crs(), mMapSettings->destinationCrs(), QgsProject::instance()->transformContext() );
+    geometry.transform( ct );
+
+    geomType = geometry.type();
+    switch ( geomType )
+    {
+      case Qgis::GeometryType::Line:
+      {
+        QgsMultiPolylineXY lines = geometry.isMultipart() ? geometry.asMultiPolyline() : QgsMultiPolylineXY() << geometry.asPolyline();
```

Comment:
```suggestion
        const QgsMultiPolylineXY lines = geometry.isMultipart() ? geometry.asMultiPolyline() : QgsMultiPolylineXY() << geometry.asPolyline();
```

Reply:
@m-kuhn , well spotted, fixed that (as well as its polygon counterpart and the rubberband code)